From f85bb3ba1220a08927bfaec360a1245547b634b2 Mon Sep 17 00:00:00 2001
From: "Valery V. Vorotyntsev" <valery_vorotyntsev@xyratex.com>
Date: Tue, 15 May 2012 23:15:31 +0300
Subject: [PATCH 182/311] confc_offline.h: start comments with a hash symbol

This style of commenting is widespread and it simplifies the parser.

LogD: #1322
RB: r/714
---
 conf/confc_offline.h | 30 ++++++++++++++----------------
 1 file changed, 14 insertions(+), 16 deletions(-)

diff --git a/conf/confc_offline.h b/conf/confc_offline.h
index 68cfb61..baf20b7 100644
--- a/conf/confc_offline.h
+++ b/conf/confc_offline.h
@@ -124,44 +124,42 @@ integer-sequence:
  *
  * UINT is an unsigned decimal integer.
  *
- * STR resembles C string; escape characters are not supported.
+ * STR resembles C string, but escape characters are not supported.
  *
  * @code
  * "Hello, world!"    -- valid STR
  * "one\ttwo\03\0\xf" -- not a STR
  * @endcode
  *
- * Tokens can be separated by white space; this consists of comments
- * and white-space characters.  Comments are surrounded by curly
- * brackets.
+ * Tokens can be separated by white-space characters.
+ * Comments start with a hash symbol and run to the end of line.
  *
  * <!---------------------------------------------------------------->
  * @subsection conf-fspec-preload-string-examples Examples
  *
 @verbatim
-                {filesystem}
+#               filesystem
 (prof:"test-2", "c2t1fs")
 
-              {rootfid} {params}    {services}
+#             rootfid   params      services
 (fs:"c2t1fs", [11, 22]  [50,60,70], ["mds", "io"])
 
-              {filesystem} {type} {end points} {node}
-(svc:"mds", "c2t1fs",      1,     ["addr0"],   "N")
+#           filesystem     type   end points                 node
+(svc:"mds", "c2t1fs",      1,     ["addr0"],                 "N")
+(svc:"io",  "c2t1fs",      2,     ["addr1","addr2","addr3"], "N")
 
-(svc:"io","c2t1fs",2,{inline comment}["addr1","addr2","addr3"],"N")
-
-           {services}     {RAM} {#CPU} {state} {flags} {pool} {nics}
-(node:"N", ["mds", "io"], 8000, 2,     3,      2,      0,     ["nic0"],
-           {sdevs}
+#          services       RAM   nr.CPU  state   flags   pool   nics
+(node:"N", ["mds", "io"], 8000, 2,      3,      2,      0,     ["nic0"],
+#          sdevs
            ["sdev0"])
 
-             {node} {type} {MTU} {speed} {filename} {state}
+#            node   type   MTU   speed   filename   state
 (nic:"nic0", "N",   5,     8192, 10000,  "ib0",     3)
 
-               {node} {iface} {media} {size}        {state} {flags} {parts}
+#              node   iface   media   size          state   flags   parts
 (sdev:"sdev0", "N",   4,      1,      596000000000, 3,      4,      ["part0"])
 
-               {sdev}   {start} {size}        {idx} {type} {filename}
+#              sdev     start   size          idx   type   filename
 (part:"part0", "sdev0", 0,      596000000000, 0,    7,     "sda1")
 @endverbatim
  *
-- 
1.8.3.2

